src/inmanta/data/model.py

```diff
@@ -308,9 +308,30 @@ class ResourceMinimal(BaseModel):
 
     id: ResourceVersionIdStr
 
+    @classmethod
+    @validator("id")
+    def name_must_contain_space(cls, v):
+        from inmanta.resources import Id  # prevent circular import
```

Comment:
I'm going to have a quick look to see if we can prevent this loop instead of importing locally. I'll get back to you.

Reply:
This seems to do it:
```diff
diff --git a/src/inmanta/data/model.py b/src/inmanta/data/model.py
index 5de29173..d00d67ce 100644
--- a/src/inmanta/data/model.py
+++ b/src/inmanta/data/model.py
@@ -28,7 +28,7 @@ from pydantic.fields import ModelField
 
 import inmanta
 import inmanta.ast.export as ast_export
-from inmanta import const, protocol
+from inmanta import const, protocol, resources
 from inmanta.stable_api import stable_api
 from inmanta.types import ArgumentTypes, JsonType, SimpleTypes, StrictNonIntBool
 
@@ -311,9 +311,7 @@ class ResourceMinimal(BaseModel):
     @classmethod
     @validator("id")
     def name_must_contain_space(cls, v):
-        from inmanta.resources import Id  # prevent circular import
-
-        if Id.is_resource_version_id(v):
+        if resources.Id.is_resource_version_id(v):
             return v
         raise ValueError(f"id {v} is not of type ResourceVersionIdStr")
 
@@ -325,9 +323,7 @@ class ResourceMinimal(BaseModel):
         takes a resource and return the same resource with it version incremented
         (the input resource is modified)
         """
-        from inmanta.resources import Id  # prevent circular import
-
-        old_res = Id.parse_id(self.id)
+        old_res = resources.Id.parse_id(self.id)
         new_res = Id(old_res.entity_type, old_res.agent_name, old_res.attribute, old_res.attribute_value, old_res.version + 1)
         self.id = new_res.resource_version_str()
         return self
```

Reply:
(`git apply <file-with-the-diff>`)